### PR TITLE
[terra-base] Update docs to use MDX syntax

### PR DESCRIPTION
**Pull request by pranav300**
### Summary
Updated base docs to use MDX syntax.

### CODE REVIEW

File: packages/terra-base/src/terra-dev-site/doc/base/Base.1.doc.mdx [hunk 1]
```diff
@@ -77,3 +85,10 @@ const locale = (navigator.languages && navigator.languages[0])
 * [Localization Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#internationalization-i18n-support)
 * [react_on_rails Compatible](https://github.com/shakacode/react_on_rails/blob/8cb06ed35cb5c2c453bcc193282b4c091574c1b7/docs/basics/i18n.md#how-to-add-i18n)
 * [CND Compatible](https://github.com/webpack/docs/wiki/configuration#outputpublicpath)
+
```
Comment (emilyrohrbough) [inline packages/terra-base/src/terra-dev-site/doc/base/Base.1.doc.mdx]: Does the AutoGenerated Peer dependency script still work with this change?

File: packages/terra-base/src/terra-dev-site/doc/base/ChangeLog.2.doc.mdx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import ChangeLog from '../../../../CHANGELOG.md';
```
Comment (emilyrohrbough) [inline packages/terra-base/src/terra-dev-site/doc/base/ChangeLog.2.doc.mdx]: Should this render the Badge as well?

File: packages/terra-base/src/terra-dev-site/doc/base/Base.1.doc.mdx (hunk 1 above)
Comment (pranav300) [reply packages/terra-base/src/terra-dev-site/doc/base/Base.1.doc.mdx]: No, it won't work with this change.

File: packages/terra-base/src/terra-dev-site/doc/base/ChangeLog.2.doc.mdx (hunk 2 above)
Comment (pranav300) [reply packages/terra-base/src/terra-dev-site/doc/base/ChangeLog.2.doc.mdx]: Updated here https://github.com/cerner/terra-core/pull/2803/commits/4e5536321edd517b2095c11caf3ed29c8e5ee657.

File: packages/terra-base/src/terra-dev-site/doc/base/Base.1.doc.mdx [hunk 3]
```diff
@@ -1,3 +1,11 @@
+import { Badge } from '../../../../package.json?dev-site-package';
```
Comment (mjhenkes) [inline packages/terra-base/src/terra-dev-site/doc/base/Base.1.doc.mdx]:
```suggestion
import { Badge } from 'terra-base/package.json?dev-site-package';
```

**Pull request merged**